inspire_mitmproxy/interaction.py

```diff
@@ -138,3 +156,33 @@ def execute_callbacks(self):
                 request=request,
                 delay=callback.get('delay', self.DEFAULT_CALLBACK_DELAY)
             )
+
+    @classmethod
+    def get_next_sequence_number_in_dir(cls, directory: Path) -> int:
+        def get_sequence_number(name) -> Optional[int]:
+            match = cls.DEFAULT_NAME_MATCH_REGEX.match(name)
+            return int(match.group(1)) if match else None
+
+        interaction_names = [
+            path.stem for path in directory.iterdir() if path.is_file() and path.suffix == '.yaml'
+        ]
+
+        sequence_numbers = sorted(filter(None.__ne__, map(get_sequence_number, interaction_names)))
```

Comment:
```python
[
    _extract_seq_number(interaction_name)
    for interaction_name in interaction_names
    if _extract_seq_number(interaction_name) is not None     
]
```

Reply:
```python
next_seq_number = 0
for interaction_name in directory.iterdir():
    if not path.is_file() or path.suffix != '.yaml':
        continue

    seq_number_match = cls.DEFAULT_NAME_MATCH_REGEX.match(interaction_name)
    if not seq_number_match:
        continue

    cur_seq_number = int(seq_number_match.group(1))
    if cur_seq_number >= next_seq_number:
       next_seq_number = cur_seq_number + 1
```

```python
def _get_seq_number_from_path(path):
    if not path.is_file() or path.suffix != '.yaml':
        return 0

    seq_number_match = cls.DEFAULT_NAME_MATCH_REGEX.match(interaction_name)
    if not seq_number_match:
        return 0

    cur_seq_number = int(seq_number_match.group(1))
    return cur_seq_number

next_seq_number = 0
for interaction_path in directory.iterdir():
    cur_seq_number = _get_seq_number_from_path(interaction_path)
    if cur_seq_number >= next_seq_number:
       next_seq_number = cur_seq_number + 1
```

Reply:
The above will fail in the case of a folder with irrelevant files: the number we get is one instead of zero, so I'll readjust this a little :+1: